libc/test/src/__support/CPP/mutex_test.cpp

```diff
@@ -0,0 +1,67 @@
+//===-- Unittests for mutex -----------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/__support/CPP/mutex.h"
+#include "test/UnitTest/Test.h"
+
+using LIBC_NAMESPACE::cpp::adopt_lock;
+using LIBC_NAMESPACE::cpp::lock_guard;
+
+static const int SIGABRT = 6;
```

Comment:
```suggestion
#include "include/llvm-libc-macros/signal-macros.h"
```

Reply:
And will need to add `libc.include.llvm-libc-macros.signal_macros` to the `DEPENDS` for this test.

Reply:
Actually, I think it would be simpler to just tracked with a simple boolean if your Mutex has been "double locked" and assert that that's the case rather than deal with signals and dest tests in this unit test.  Is that doable?